### PR TITLE
Restore JDBC Kerberos authentication behavior for UGI.doAs

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

A typical use case of Hadoop UGI w/ Kyuubi Hive JDBC is
```
UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
ugi.doAs(() -> {
  Connection conn = DriverManager.getConnection(
        "jdbc:kyuubi://host:10009/default;principal=kyuubi@_HOST/ABC.ORG");
  ...
});
```

After https://github.com/apache/kyuubi/pull/3023, Kyuubi Hive JDBC implements the Kerberos authentication by using JDK directly instead of Hadoop `UserGroupInformation`, but it also introduce a breaking change for Hadoop users, including the above case. As workaround, user should add `kerberosAuthType=fromSubject` alongside w/ `principal=kyuubi@_HOST/ABC.ORG` to make it work.

This PR propose to restore the behavior before https://github.com/apache/kyuubi/pull/3023 by handling UGI.doAs explicitly.

And this PR makes the `clientPrincipal` `clientKeytab` as the highest priority, so in below cases, `clientPrincipal` `clientKeytab` take effects instead of UGI.

```
UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
ugi.doAs(() -> {
  Connection conn = DriverManager.getConnection(
        "jdbc:kyuubi://host:10009/default;principal=kyuubi@_HOST/ABC.ORG;" + 
        "clientPrincipal=tom@_HOST/ABC.ORG;clientKeytab=/path/xxx.keytab");
  ...
});
```

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by cxzl25 — APPROVED** [review]

**Comment by pan3793** [comment]
Have a offline talk w/ @gabrywu, this PR fixed the original issue but also brings a new one, considering the following case, the `clientPrincipal` and `clientKeytab` will be totally ignored and always respect UGI, personally I think it's an acceptable result, but would like to know other guys thoughts.

cc @cxzl25 @zhouyifan279 @turboFei 

```
UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
ugi.doAs(() -> {
  Connection conn = DriverManager.getConnection(
        "jdbc:kyuubi://host:10009/default;principal=kyuubi@_HOST/ABC.ORG;" + 
        "clientPrincipal=tom@_HOST/ABC.ORG;clientKeytab=/path/xxx.keytab");
  ...
});
```

**Review by turboFei — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]
LGTM

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java [hunk 1]
```diff
@@ -30,10 +30,7 @@
 import java.net.UnknownHostException;
 import java.nio.ByteBuffer;
 import java.nio.charset.StandardCharsets;
-import java.security.AccessControlContext;
-import java.security.AccessController;
-import java.security.KeyStore;
-import java.security.SecureRandom;
+import java.security.*;
```
Comment (merrily01) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:33]: Why not introduce a specific reference, but modify it to * ?

Comment (pan3793) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:33]: it's the result of `dev/reformat`, part of Google Java Format standard

Comment (merrily01) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:33]: ok, thx

**Comment by turboFei** [comment]
thanks, merged to master and 1.7

**Pull request closed without merging**